src/openqaoa-core/algorithms/qaoa/qaoa_workflow.py

```diff
@@ -250,12 +251,24 @@ def compile(
             total_annealing_time=self.circuit_properties.annealing_time,
         )
 
+        backend_dict = self.backend_properties.__dict__.copy()
+
         self.backend = get_qaoa_backend(
             qaoa_descriptor=self.qaoa_descriptor,
             device=self.device,
-            **self.backend_properties.__dict__,
+            **backend_dict,
         )
 
+        if (
```

Comment:
Let's add some comments here for future reference.

Reply:
Done. Please review.